### PR TITLE
config-upload, and set env=host in harpoon unit file

### PR DESCRIPTION
**Pull request by sallyom**
This PR introduces the **dynamic reloading of targets**.
Ways for user to set and/or update harpoon config file, which will prompt a reload:
*  A ConfigTarget with a URL (raw GH file) set in the `initial config file on disk` at `~/.harpoon/config.yaml` 
* `$HARPOON_CONFIG_URL` env var set within the harpoon pod - with a podman command
 `podman run -e  HARPOON_CONFIG_URL=blah/config.yaml  - - - quay.io/harpoon/harpoon:latest`
* If both of those are set the configTarget.Url from an existing config file will be used, and will set the env var to match the target found from a config file.
* If `$HARPOON_CONFIG_URL` is set and _no ConfigTarget_ is set in the config file that the URL points to, harpoon will process with the env variable with the intial startup, but will not schedule any config update jobs.

**In summary:**
```
You can start with no config file on disk if you have $HARPOON_CONFIG_URL 
set in the harpoon pod (most likely from a podman run cmd, and most likely from the harpoon unit file).
If the config file that is downloaded from the config URL does not have a ConfigTarget, harpoon will not 
schedule config updates.
```

**Upon reloading of a new config:**
It's a hard reset, new targets replace oldTargets. This works, because, with a podman-volume intact, you'll have your last known commit already cloned and fetched, so the new jobs will pick up where they left off. Harpoon will set the LastCommit for all targets that were pre-existing to the last known commit for that target/method.

### CODE REVIEW

**Comment by sallyom** [comment]
@cooktheryan @josephsawaya this is ready for review now :grin:

**Comment by cooktheryan** [comment]
Almost all green go @sallyom go!!!!!

**Comment by sallyom** [comment]
> Almost all green go @sallyom go!!!!!

:grin: :green_book:

**Comment by sallyom** [comment]
@cooktheryan fixing up how to find the config file, then this will be ready to go - the `--config` flag actually does nothing, so I'll remove that (only way it makes any sense is if you `./harpoon start --config` and we don't do that.

**Comment by sallyom** [comment]
@cooktheryan I added a workflow for the harpoon service & the config reload, but I put an 8min wait because of the 5min default GH load-time of raw URLs - We can avoid this by serving the file elsewhere. wdyt?

**Comment by cooktheryan** [comment]
@sallyom i'm trying to think of external places to run it but with the 8 minute wait maybe we make the job run only on main and allow the subsequent jobs to be triggered as well (like the arm builds) to make the most bang for our buck. I don't think there is a problem though having the 8 minute wait in there because it definitely allows us to test customer functionality.

**Comment by cooktheryan** [comment]
@sallyom we could potentially remove the sleep and do either and until or while script to check for the changed output vs the original started output. That would allow for us to have a more intelligent checker than the sleeps and allow us to move cleaner and faster.

**Comment by sallyom** [comment]
@cooktheryan I added the config-reload job without harpoon unit file/service for this PR. As a followup, we can add a workflow for the harpoon service.

File: pkg/engine/harpoon.go [hunk 1]
```diff
@@ -740,104 +927,67 @@ func (hc *HarpoonConfig) getGit(target *api.Target, initialRun bool) (*object.Co
...
-	}
-
-	// ... retrieve the tree from the commit
-	tree, err := newCommit.Tree()
-	if err != nil {
-		return nil, nil, fmt.Errorf("error when retrieving tree: %s", err)
+		klog.Info(err)
+		// if initial, we're here because could not read/find local config
+		if initial {
+			klog.Infof("HERERRE ERRRRROR: %v", err)
```
Comment (josephsawaya) [inline pkg/engine/harpoon.go:988]:
```suggestion

```
